### PR TITLE
workflows: moves remaining files

### PR DESCRIPTION
**Pull request by jalavik**
- Moves old BibWorkflow files to new workflows module.
- Updates imports in deposit module.

### CODE REVIEW

**Comment by jalavik** [comment]
Changing the import to `invenio.base.globals import cfg` caused a cascade of errors while loading the page:

```
ERROR in flask_registry [/home/jlavik/envs/nextstructure/src/nextstructure/invenio/ext/registry/flask_registry.py:324]:
Could not import: "invenio.modules.deposit.views: working outside of application context
```

Looking into why..

**Comment by jalavik** [comment]
@jirikuncar:

Javascript inclusions and URLs fixed as you proposed. Still getting the blueprint import error though, could it be a problem in flask_registry?

**Comment by jirikuncar** [comment]
When you are more or less done, we can push work-in-progress commits (don't forget to rebase).

**Comment by jirikuncar** [comment]
Can you add tests `self.client.get(url_for('workflows.*'))` for pages that should be accessible even without demosite? See `invenio.modules.search.testsuite` ... just not to break them again. Thanks

File: invenio/modules/deposit/models.py [hunk 1]
```diff
@@ -33,9 +33,9 @@
 from flask.ext.login import current_user
 
 from invenio.ext.sqlalchemy import db
-from invenio.bibworkflow_config import CFG_OBJECT_VERSION, CFG_WORKFLOW_STATUS
+from invenio.modules.workflows.config import CFG_OBJECT_VERSION, CFG_WORKFLOW_STATUS
```
Comment (jirikuncar) [inline invenio/modules/deposit/models.py:36]: :(

File: invenio/modules/deposit/views/api.py [hunk 2]
```diff
@@ -24,7 +24,7 @@
     jsonify
 
 from ..loader import deposition_types
-from invenio.bibworkflow_config import CFG_WORKFLOW_STATUS
+from invenio.modules.workflows.config import CFG_WORKFLOW_STATUS
```
Comment (jirikuncar) [inline invenio/modules/deposit/views/api.py:27]: :(

File: invenio/modules/workflows/api.py [hunk 3]
```diff
@@ -267,8 +267,8 @@ def resume_objects_in_workflow(id_workflow, start_point="continue_next",
 
     @yield: BibWorkflowEngine that ran the workflow
     """
-    from invenio.modules.workflows.models import BibWorkflowObject
-    from invenio.bibworkflow_config import CFG_OBJECT_VERSION
+    from .models import BibWorkflowObject
+    from .config import CFG_OBJECT_VERSION
```
Comment (jirikuncar) [inline invenio/modules/workflows/api.py:271]: :(

File: invenio/modules/workflows/client.py [hunk 4]
```diff
@@ -18,8 +18,8 @@
 
 import traceback
 from workflow.engine import HaltProcessing
-from invenio.bibworkflow_config import CFG_OBJECT_VERSION
-from invenio.bibworkflow_config import CFG_WORKFLOW_STATUS
+from .config import CFG_OBJECT_VERSION
+from .config import CFG_WORKFLOW_STATUS
```
Comment (jirikuncar) [inline invenio/modules/workflows/client.py:23]: :(

File: invenio/modules/workflows/engine.py [hunk 5]
```diff
@@ -29,17 +27,16 @@
...
-from invenio.bibworkflow_config import (CFG_WORKFLOW_STATUS,
-                                        CFG_OBJECT_VERSION)
-from invenio.modules.workflows.logger import (get_logger,
-                                              BibWorkflowLogHandler)
+from .models import (Workflow,
+                     BibWorkflowObject,
+                     BibWorkflowEngineLog)
+from .utils import dictproperty, get_workflow_definition
+from .config import (CFG_WORKFLOW_STATUS,
+                     CFG_OBJECT_VERSION)
```
Comment (jirikuncar) [inline invenio/modules/workflows/engine.py:37]: :(

File: invenio/modules/workflows/models.py [hunk 6]
```diff
@@ -28,8 +28,8 @@
 from sqlalchemy.orm.exc import NoResultFound
 from invenio.ext.sqlalchemy import db
 from invenio.base.globals import cfg
-from invenio.bibworkflow_config import CFG_OBJECT_VERSION
-from invenio.bibworkflow_utils import redis_create_search_entry
+from .config import CFG_OBJECT_VERSION
```
Comment (jirikuncar) [inline invenio/modules/workflows/models.py:31]: :(

File: invenio/modules/workflows/static/js/workflows/entry_details.js [hunk 7]
```diff
@@ -17,13 +17,14 @@
...
-if ( window.addEventListener ) {
-        $("div.btn-group[name='object_preview_btn']").bind('click', function(event){
+function bind_object_preview(url_prefix, entry_id) {
+    $("div.btn-group[name='object_preview_btn']").bind('click', function(event){
           var format = event.target.name;
           jQuery.ajax({
-                url: url_entry_data_preview + "?oid={{ entry.id }}&format=" + format,
+                url: url_prefix,
+                data: {'oid': entry_id,
+                       'format': format},
```
Comment (jirikuncar) [inline invenio/modules/workflows/static/js/workflows/entry_details.js:27]: :+1:

File: invenio/modules/workflows/worker_engine.py [hunk 8]
```diff
@@ -17,10 +17,10 @@
...
 
 from invenio.ext.sqlalchemy import db
-from invenio.bibworkflow_client import run_workflow, continue_execution
-from invenio.bibworkflow_engine import BibWorkflowEngine
-from invenio.modules.workflows.models import BibWorkflowObject, Workflow
-from invenio.bibworkflow_config import CFG_OBJECT_VERSION
+from .client import run_workflow, continue_execution
+from .engine import BibWorkflowEngine
+from .models import BibWorkflowObject, Workflow
+from .config import CFG_OBJECT_VERSION
```
Comment (jirikuncar) [inline invenio/modules/workflows/worker_engine.py:23]: :(

**Comment by jirikuncar** [comment]
It's getting there ... config needed ;)

**Comment by jalavik** [comment]
Sure, but I mentioned already that its not really a config, but its constants (as was usual in legacy invenio), so I asked if I should move it to a constants.py or distribute them in engine.py and models.py

**Comment by jirikuncar** [comment]
What about this?

```
class BibWorkflowObject:
    CFG_OBJECT_VERSION = enum(...)
```

**Comment by jalavik** [comment]
Sure, we can do that, that is adding it to BibWorkflowObject and BibWorkflowEngine class definitions. However, since changing this everywhere is a big, tedious change touching several places that liekly will cause conflict galore with the upcoming big code branch from @PX9e, I would suggest that we move forward with what we got right now. 

Then as soon as the branch backlog for workflows/holdingpen is taken care of we will fix this. Don't worry, this will be on the top of our todo-list after.

**Comment by jalavik** [comment]
About adding the testsuite, will it work with restricted pages under /admin? Also, I just tried adding a simple test, but it fails with:

```
======================================================================
FAIL: test_main_admin_availability (invenio.modules.workflows.testsuite.test_workflows.WorkflowViewTest)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/jlavik/envs/nextstructure/src/nextstructure/invenio/modules/workflows/testsuite/test_workflows.py", line 32, in test_main_admin_availability
    self.assert200(response)
  File "/home/jlavik/envs/nextstructure/local/lib/python2.7/site-packages/flask_testing.py", line 192, in assert200
    self.assertStatus(response, 200)
  File "/home/jlavik/envs/nextstructure/local/lib/python2.7/site-packages/flask_testing.py", line 181, in assertStatus
    self.assertEqual(response.status_code, status_code)
AssertionError: 401 != 200
-------------------- >> begin captured logging << --------------------
invenio.base: ERROR: Missing files: /yourtags/static/css/tags/popover.css
--------------------- >> end captured logging << ---------------------
```

I have seen this file missing in the console several times. Is it me or is it a common occurrence?

**Comment by jirikuncar** [comment]
If you should get 401 then we can just change the test case to `assert401` and I will merge it as-it-is. I note that you will change the `config` later.

**Pull request merged**